packages/ui/src/theme/createTheme.ts

```diff
@@ -74,6 +82,53 @@ export function createTheme(
   theme?: Theme,
   baseTheme: BaseTheme = defaultTheme
 ): WebTheme {
+  /*
+1. find path to deprecated values in baseTheme
+2. see if values exist in theme that match depricated paths from base
+3. determine undepricated path
+4. see if values in theme exist for undepricated paths
+5. if no value of undepricated exists then replace it with value of depricated
+ */
+  if (theme) {
+    let deprecated: pathObject[] = findDeprecatedStateTokens(baseTheme);
+
+    deprecated.forEach((element: pathObject) => {
+      //look for a deprecated value in the theme
+      const path: string[] = element.path;
+      const deprecatedThemeValue = pathToValue(theme, path);
+
+      //use the deprecated path to generate the new path by replacing any state names with _{state}
+      const targetPath = path.map((pathValue) => {
+        if (STATE_NAMES.includes(pathValue)) {
+          return `_${pathValue}`;
+        }
+        return pathValue;
+      });
+
+      //follow the path to determine if a value already exists within the theme for the non deprecated version
+      let themeValue = theme.tokens || {};
+      targetPath.every((pathValue) => {
```

Comment:
What's the need for using `every` here? Why not just `forEach`?

Reply:
every just gives us an easy way to drop out of the iteration once the condition fails.

Reply:
How about using `find`? 

something like...

```
themeValue = targetPath.find(pathValue => themeValue[pathValue])
```